apiv2/apiv2_utils.py

```diff
@@ -256,6 +257,18 @@ def initial(self, request, *args, **kwargs):
         self.get_request_information(request)
         self.store_monitor_usage()
 
+    def get_serializer(self, *args, **kwargs):
+        # Overwrite DRF's get_serializer method so that we add sound analysis information when requested
+        serializer_class = self.get_serializer_class()
+        kwargs['context'] = self.get_serializer_context()
+        if 'SoundListSerializer' in str(serializer_class):
+            # If we are trying to serialize sounds, check if we should and sound analysis data to them and add it
+            if type(args[0] == list):
```

Comment:
`list` might be a bit restrictive. anything iterable should work.
maybe we can use 
`isinstance(args[0], Iterable)` instead?

Reply:
Thanks, I'll update that :)